### PR TITLE
Added implementation of AVL tree in PHP

### LINKED ISSUE #2112: AVL tree
- [x] C
- [x] C++ - @Manish-cloud
- [ ] C# - @KanakalathaVemuru
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [ ] JS - @satvikchachra
- [ ] Kotlin
- [ ] Php - @chandrikadeb7
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by chandrikadeb7**
Fixes #2112 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added AVL Tree implementation in PHP
- 

#### Languages Used:
- PHP
- 

#### Files Added:
- avl_tree.php
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by chandrikadeb7** [comment]
@singh-shreya6 Please review.

File: AVL_Tree/avl_tree.php [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        $ldepth = $this->left !== NULL
+                ? $this->left->depth
+                : 0;
+               
+        $rdepth = $this->right !== NULL
+                ? $this->right->depth
+                : 0;
+               
+               
+        if( $ldepth > $rdepth+1 )
```
Comment (singh-shreya6) [inline AVL_Tree/avl_tree.php:29]: Space around +

File: AVL_Tree/avl_tree.php [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                     : 0;
+
+            if( $lldepth < $lrdepth ) 
+            { // LR rotation
+                $this->left->rotateRR(); // consist of a RR rotation of the left child ...
+            } // ... plus a LL rotation of this node, which happens anyway
+
+            $this->rotateLL();
+        } 
+        else if( $ldepth+1 < $rdepth )
```
Comment (singh-shreya6) [inline AVL_Tree/avl_tree.php:46]: Space around +

File: AVL_Tree/avl_tree.php [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        if( $this->right !== NULL && $this->depth <= $this->right->depth )
+            $this->depth = $this->right->depth+1;
+    }
+
+    // --- debugging functions
+       
+    function toString() 
+    {
+        $s = "<table border><tr>\n".$this->toTD(0)."</tr>\n";
+        $depth = $this->depth-1;
```
Comment (singh-shreya6) [inline AVL_Tree/avl_tree.php:117]: Space around -

File: AVL_Tree/avl_tree.php [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+           
+            if( $this->right !== NULL )
+                $nleafs += $this->right->getNLeafs();
+            else
+                ++$nleafs; // lus one for the right "stump"
+        } 
+        else 
+        {
+            if( $this->right !== NULL )
+                $nleafs = $this->right->getNLeafs()+1; // plus one for the left "stump"
```
Comment (singh-shreya6) [inline AVL_Tree/avl_tree.php:183]: Space around +

**Pull request merged**